aicspylibczi/CziFile.py

```diff
@@ -139,68 +148,256 @@ def pixel_type(self):
+            bbox.h = The height of the bounding box.
 
         """
-        bbox = self.reader.read_scene_wh(index)
-        ans = (bbox.x, bbox.y, bbox.w, bbox.h)
-        return ans
+        plane_constraints = self._get_coords_from_kwargs(kwargs)
+        dims, bbox = self.reader.read_tile_bounding_box(plane_constraints)
+        return bbox
 
-    def scene_height_by_width(self, index: int = -1):
+    def get_scene_bounding_box(self, index: int = 0):
         """
-        Get the size of the scene (Y, X) / (height, width) for the specified Scene. The default is to return
-        the size of the first Scene but Zeiss allows scenes to be different sizes thought it is unlikely to encounter.
-        This module will warn you on instantiation if the scenes have inconsistent width and height.
+        Get the bounding box (pyramid=0) for the specified scene. For non-mosaic files.
+        This should be equivalent to the results from get_tile_bounding_box but requiring only one arguments.
 
         Parameters
         ----------
         index
-            specifies the index of the Scene to get the height and width of
+             the scene index, if omitted it defaults to Zero
 
         Returns
         -------
-        tuple
-            (height, width) tuple of the Specified scene.
+        bbox
+             A Bounding Box, bbox, of type BBox.
+             bbox.x = The x coordinate of the upper left corner of the bounding box.
+             bbox.y = The y coordinate of the upper left corner of the bounding box.
+             bbox.w = The width of the bounding box.
+             bbox.h = The height of the bounding box.
+
+        """
+        bbox = self.reader.read_scene_bounding_box(index)
+        return bbox
+
+    def get_all_tile_bounding_boxes(self, **kwargs):
+        """
+        Get one or more tiles (subblocks) bounding boxes (pyramid=0) for the specified dimensions.
+        For non-mosaic files.
+
+        Parameters
+        ----------
+        kwargs
+            The keywords below allow you to specify the dimensions that you wish to match. If you
+            under-specify the constraints you can easily end up with a massive image stack.
+                       Z = 1   # The Z-dimension.
+                       C = 2   # The C-dimension ("channel").
+                       T = 3   # The T-dimension ("time").
+                       R = 4   # The R-dimension ("rotation").
+                       S = 5   # The S-dimension ("scene").
+                       I = 6   # The I-dimension ("illumination").
+                       H = 7   # The H-dimension ("phase").
+                       V = 8   # The V-dimension ("view").
+
+        Returns
+        -------
+        dict[tile_info, bbox]
+            A dictionary with keys of type TileInfo and values of type BBox.
+            For a key, ie tle, of type Tile:
+                tle.dimension_coordinates = A dictionary of Dimensions for the tile
```

Comment:
typo on `tle` -> `tile`? or is it actually called `tle`?

Reply:
tle is just a variable name I was worried about using the word tile and a variable named tile in the same block so I mutated it. Would it be better if it were `tile.dimension_coordinates = A dictionary of Dimensions for the tile/subblock`?

Reply:
Yea I think it is nitpicky of me but would make more sense.